content-src/lib/getVideoPreview.js

```diff
@@ -0,0 +1,36 @@
+module.exports = function getVideoPreview(site) {
+  switch (site.provider_url) {
+    case "https://www.youtube.com/":
```

Comment:
We've also used [**get-youtube-id**](https://www.npmjs.com/package/get-youtube-id) in other projects (such as [**meandavejustice/min-vid**](https://github.com/meandavejustice/min-vid)).

Not sure if there is a similar library to try and parse URLs and extract vimeo ids. I did find [**vimeo-regex**](https://github.com/regexhq/vimeo-regex), but not sure how much it will help vs bloat.